File: agent/query_engine/LegacyEngineStrategy.py
from __future__ import annotations

from typing import AsyncIterator, Callable

from query_engine.EngineContext import EngineContext


class LegacyEngineStrategy:
    def __init__(self, stream_fn: Callable[..., AsyncIterator[str]]) -> None:
        self._stream_fn = stream_fn

    async def run(self, context: EngineContext) -> AsyncIterator[str]:
        async for event in self._stream_fn(
            context.messages,
            user_id=context.user_id,
            session_id=context.session_id,
            trace_id=context.trace_id,
            turn_id=context.turn_id,
        ):
            yield event

File: agent-ts/src/query_engine/LegacyEngineStrategy.ts
import type { EngineContext } from "./EngineContext.js";
import type { EngineEvent } from "./EngineEvent.js";
import { parseSseToEngineEvent } from "./parse_sse_to_engine_event.js";

export class LegacyEngineStrategy {
  constructor(private readonly streamFn: (messages: EngineContext["messages"], context: EngineContext) => AsyncIterable<string>) {}

  async *run(context: EngineContext): AsyncIterable<EngineEvent> {
    for await (const rawEvent of this.streamFn(context.messages, context)) {
      yield parseSseToEngineEvent(rawEvent);
    }
  }
}
